openmls/src/group/tests/test_proposal_validation.rs

```diff
@@ -947,5 +1029,837 @@ fn test_valsem105(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider
+
+    assert_eq!(
+        err,
+        UnverifiedMessageError::InvalidCommit(StageCommitError::ProposalValidationError(
+            ProposalValidationError::CommitterIncludedOwnUpdate
+        ))
+    );
+
+    let original_update_plaintext =
+        VerifiableMlsPlaintext::tls_deserialize(&mut serialized_update.as_slice())
+            .expect("Could not deserialize message.");
+
+    // Positive case
+    let unverified_message = bob_group
+        .parse_message(MlsMessageIn::from(original_update_plaintext), backend)
+        .expect("Could not parse message.");
+    bob_group
+        .process_unverified_message(unverified_message, None, backend)
+        .expect("Unexpected error.");
+}
+
+/// ValSem112
+/// Update Proposal:
+/// The sender of a standalone update proposal must be of type member
+#[apply(ciphersuites_and_backends)]
+fn test_valsem112(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider) {
+    // Before we can test creation or reception of (invalid) proposals, we set
+    // up a new group with Alice and Bob.
+    let ProposalValidationTestSetup {
+        mut alice_group,
+        mut bob_group,
+    } = validation_test_setup(*PURE_PLAINTEXT_WIRE_FORMAT_POLICY, ciphersuite, backend);
+
+    // This can really only be tested by the receiver, as there is no way to
+    // make a client create a proposal with a different sender type than
+    // `member`.
+
+    // However, we can test the receiving side by crafting such a proposal
+    // manually.
+    let commit = alice_group
+        .propose_self_update(backend, None)
+        .expect("Error creating self-update");
+
+    // Check that the sender type is indeed `member`.
+    let serialized_update = commit
+        .tls_serialize_detached()
+        .expect("error serializing plaintext");
+
+    let plaintext = VerifiableMlsPlaintext::tls_deserialize(&mut serialized_update.as_slice())
+        .expect("Could not deserialize message.");
+
+    assert!(plaintext.sender().is_member());
+
+    let mut plaintext = VerifiableMlsPlaintext::tls_deserialize(&mut serialized_update.as_slice())
+        .expect("Could not deserialize message.");
+
+    // Keep the original plaintext for positive test later.
+    let original_plaintext = plaintext.clone();
+
+    // Now let's change the sender type to NewMember.
+    plaintext.set_sender(Sender::NewMember);
+
+    let update_message_in = MlsMessageIn::from(plaintext.clone());
+
+    // Have bob process the resulting plaintext
+    let err = bob_group
+        .parse_message(update_message_in, backend)
+        .expect_err("Could parse unverified message despite modified public key in path.");
+
+    assert_eq!(
+        err,
+        ParseMessageError::ValidationError(ValidationError::NotACommit)
+    );
+
+    // We can't test with sender type Preconfigured, since that currently panics
+    // with `unimplemented`.
+    // TODO This test should thus be extended when fixing #106.
```

Comment:
I guess we don't need to change the code as such. It's just that `ValSem112` is not according to spec because of that. If you can find a suitable place to mention that somewhere, it would be nice (but it's not a blocker).